extension/healthcheckextension/healthcheckextension.go

```diff
@@ -35,9 +37,20 @@ var _ (extension.PipelineWatcher) = (*healthCheckExtension)(nil)
 func (hc *healthCheckExtension) Start(host extension.Host) error {
 
 	hc.logger.Info("Starting health-check extension", zap.Any("config", hc.config))
+
```

Comment:
Why is the healthCheckExtension changed in a PR that is about Bump Jaeger version? Are these 2 things related in any way? If not please split, we want one PR per logical change.

Reply:
The `healthcheck` package in jaeger has changed since v1.9.0 (http.Server was moved out of Healthcheck struct). These changes in `healthcheckextension` fix the following compile time errors -

```
# github.com/open-telemetry/opentelemetry-service/extension/healthcheckextension
extension/healthcheckextension/healthcheckextension.go:40:25: hc.server.Serve undefined (type *healthcheck.HealthCheck has no field or method Serve)
extension/healthcheckextension/healthcheckextension.go:66:26: too many arguments in call to healthcheck.New
extension/healthcheckextension/healthcheckextension.go:66:52: undefined: healthcheck.Logger
# github.com/open-telemetry/opentelemetry-service/extension/healthcheckextension [github.com/open-telemetry/opentelemetry-service/extension/healthcheckextension.test]
extension/healthcheckextension/healthcheckextension.go:40:25: hc.server.Serve undefined (type *healthcheck.HealthCheck has no field or method Serve)
extension/healthcheckextension/healthcheckextension.go:66:26: too many arguments in call to healthcheck.New
extension/healthcheckextension/healthcheckextension.go:66:52: undefined: healthcheck.Logger
```

Reply:
@pjanotti can you please chime in here? I wasn't aware that healtcheck extension had a direct interaction with Jaeger package. Was this intended?

Reply:
@tigrannajaryan actually this a dependency that we want to remove but we have to refactor it to our own implementation, see https://github.com/open-telemetry/opentelemetry-service/issues/307

Reply:
Aha, my bad, did not see that issue. Will first work on refactoring.

Reply:
@annanay25 I think it is OK for you to merge this PR as is since it is valuable on its own and then we can do the refactoring (feel free to do it yourself if you want). @pjanotti are you OK with this approach?

Reply:
Yes, @tigrannajaryan and @annanay25 let's merge as it is and do the refactor in a separate PR.

Reply:
Thanks!